### PR TITLE
Sorting Algoritms

### PR DESCRIPTION
**Pull request by RahulKakarla**
wrote sorting algorithms based on my understanding

### CODE REVIEW

File: Java/Rahul/BubbleSort.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+public class BubbleSort{
```
Comment (manideeppabba1991) [inline Java/Rahul/BubbleSort.java:1]: Please mention time & space complexities

**Comment by RahulKakarla** [comment]
Sure will do that in sometime, just close that laptop

File: Java/Rahul/InsertionSort.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+     }
+     
+     public static int[] insertionSort(int[] input){
+         
+        for (int j=1 ;j<input.length;j++){
+            for(int i = j ; i > 0 ;i--){
+                if(input[i] < input[i-1]){
+                    int temp = input[i-1];
+                    input[i-1] = input[i];
+                    input[i] = temp;
```
Comment (manideeppabba1991) [inline Java/Rahul/InsertionSort.java:20]: your inner loop continues to compare values even after insertion is done. Can you think about ways to stop that and make your code little bit more faster ?

File: Java/Rahul/SelectionSort.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+     }
+     
+     public static int[] selectionSort(int[] input){
+         
+        for (int j=0;j<input.length-1;j++){
+            int minimum = input[j];
+            int minIndx = j;
+            for(int i= j ; i < input.length-1 ;i++){
+                if(minimum > input[i+1]){
```
Comment (manideeppabba1991) [inline Java/Rahul/SelectionSort.java:19]: why are we comparing with i+1 ? While this absolutely works fine, It is bit confusing. May be starting inner loop at j+1 will be more clearer? What do you think?

**Comment by manideeppabba1991** [comment]
Overall, Great work! you are now one step closer to reaching your goals. Way to go!

File: Java/Rahul/InsertionSort.java (hunk 2 above)
Comment (RahulKakarla) [reply Java/Rahul/InsertionSort.java:20]: Will try

File: Java/Rahul/SelectionSort.java (hunk 3 above)
Comment (RahulKakarla) [reply Java/Rahul/SelectionSort.java:19]: Yes I agree
I thought that in insertionsort and changed that
Will change this as well

File: Java/Rahul/InsertionSort.java (hunk 2 above)
Comment (RahulKakarla) [reply Java/Rahul/InsertionSort.java:20]: That is expected right?
In insertion we need to insertions until it gets to first place

Comment (manideeppabba1991) [reply Java/Rahul/InsertionSort.java:20]: It does not have to go to the first place all the time. If `current value` is greater the `i-1`, it should stop iterating and insert it into that position. This make our code little bit faster by skipping few iterations.

Comment (RahulKakarla) [reply Java/Rahul/InsertionSort.java:20]: Got it
Will change
Thank you

**Review by manideeppabba1991 — DISMISSED** [review]

File: Java/Rahul/SelectionSort.java (hunk 3 above)
Comment (RahulKakarla) [reply Java/Rahul/SelectionSort.java]: changed

**Comment by RahulKakarla** [comment]
> Overall, Great work! you are now one step closer to reaching your goals. Way to go!

thank you Mani

**Review by manideeppabba1991 — APPROVED** [review]

**Pull request merged**